File: python/500.py
class Solution:
    def findWords(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        r1 = "qwertyuiop"
        r2 = "asdfghjkl"
        r3 = "zxcvbnm"

        ans = []
        d = {}
        for r in r1:
            d[r] = 1
        for r in r2:
            d[r] = 2
        for r in r3:
            d[r] = 3

        for word in words:
            temp = word.lower()
            pre = d[temp[0]]
            same = True
            for x in temp:
                if d[x] != pre:
                    same = False
                    break
            if same:
                ans.append(word)
        return ans

File: java/500.java
class Solution {
    ArrayList<String> ans = new ArrayList();
    HashMap<Character,Integer>map = new HashMap();
    public String[] findWords(String[] words) {
        
        for(char c:"qwertyuiopQWERTYUIOP".toCharArray()){
            map.put(c,0);
        }
        for(char c:"asdfghjklASDFGHJKL".toCharArray()){
            map.put(c,1);
        }
        for(char c:"zxcvbnmZXCVBNM".toCharArray()){
            map.put(c,2);
        }
        for(String word:words){
            add(word);
        }
        return ans.toArray(new String[ans.size()]);
    }
    
    public void add(String word){
        int num = map.get(word.charAt(0));
        for(char c:word.toCharArray()){
            if(map.get(c) != num){
                return;
            }
        }
        ans.add(word);
    }
}